File: src/main/java/land/leets/domain/application/usecase/UpdateResultImpl.java
package land.leets.domain.application.usecase;

import land.leets.domain.application.domain.Application;
import land.leets.domain.application.domain.repository.ApplicationRepository;
import land.leets.domain.application.exception.ApplicationNotFoundException;
import land.leets.domain.application.presentation.dto.StatusRequest;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class UpdateResultImpl implements UpdateResult {

    private final ApplicationRepository applicationRepository;

    @Override
    public Application execute(Long id, StatusRequest request) {
        Application application = applicationRepository.findById(id).orElseThrow(ApplicationNotFoundException::new);
        request.updateApplication(application);
        return applicationRepository.save(application);
    }
}

File: src/main/kotlin/land/leets/domain/application/usecase/UpdateResultImpl.kt
package land.leets.domain.application.usecase

import land.leets.domain.application.domain.Application
import land.leets.domain.application.domain.repository.ApplicationRepository
import land.leets.domain.application.exception.ApplicationNotFoundException
import land.leets.domain.application.presentation.dto.StatusRequest
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
class UpdateResultImpl(
    private val applicationRepository: ApplicationRepository
) : UpdateResult {

    override fun execute(id: Long, request: StatusRequest): Application {
        val application = applicationRepository.findByIdOrNull(id) ?: throw ApplicationNotFoundException()
        application.updateApplicationStatus(request.applicationStatus)
        return applicationRepository.save(application)
    }
}
